test/unit/plugins/rest_spec.js

```diff
@@ -131,40 +147,180 @@ describe("The Rest plugin", function () {
 			});
 		});
 
-		describe("when there is a problem with the database connection", function () {
-			var result;
+	});
+
+	describe("creating a new instance", function () {
+		var server          = new Hapi.Server();
+		var instanceAdapter = new InstanceAdapter();
+		var createInstanceStub;
+		var runInstancesStub;
+		var updateInstanceStub;
+
+		var awsAdapter = new AwsAdapter();
+
+		before(function () {
+			createInstanceStub = Sinon.stub(instanceAdapter, "createInstance")
+			.returns(Bluebird.resolve({
+				ami   : "ami-d05e75b8",
+				type  : "t2.micro",
+				state : "pending",
+				uri   : ""
+			}));
+
+			updateInstanceStub = Sinon.stub(instanceAdapter, "updateInstance")
+			.returns(Bluebird.resolve({
+				ami   : "ami-d05e75b8",
+				type  : "t2.micro",
+				state : "failed",
+				uri   : ""
+			}));
+
+			server.connection();
 
+			return server.registerAsync({
+				register : Rest,
+				options  : {
+					awsAdapter : awsAdapter
+				}
+			});
+		});
+
+		after(function () {
+			updateInstanceStub.restore();
+			createInstanceStub.restore();
+			return server.stopAsync();
+		});
+
+		describe("when the credentials aren't properly configured", function () {
 			before(function () {
-				Sinon.stub(mapper, "create", function () {
-					return Bluebird.reject(new Error("Simulated Failure."));
+				var AuthError  = new Error();
+				AuthError.name = "AuthFailure";
+
+				runInstancesStub = Sinon.stub(awsAdapter, "runInstances")
+				.rejects(AuthError);
+			});
+
+			after(function () {
+				runInstancesStub.restore();
+			});
+
+			it("returns an error with statusCode 400", function () {
+				var request = new Request("POST", "/v1/instances").mime("application/json").payload({
+					ami  : VALID_AMI,
+					type : VALID_TYPE
 				});
+				return request.inject(server)
+				.then(function (response) {
+					expect(response.statusCode, "status").to.equal(500);
```

Comment:
The description says that the status code should be `400` while the test case checks for `500`. Which should it actually be?